contracts/Conference.sol

```diff
@@ -172,9 +177,50 @@ contract Conference is Destructible, GroupAdmin {
      * @return True if the user is marked as attended by admin.
      */
     function isAttended(address _addr) public view returns (bool){
-        return isRegistered(_addr) && participants[_addr].attended;
+        if (!isRegistered(_addr)) {
+            return false;
+        }
+
+        if (participants[_addr].attended) {
+            return true;
+        }
+
+        // now check the attendance maps
+        if (0 < attendanceMaps.length) {
+            Participant storage p = participants[_addr];
+            uint pIndex = p.participantIndex - 1;
+            uint map = attendanceMaps[uint(pIndex / 256)];
+            return (0 < (map & (2 ** (pIndex % 256))));
+        }
+
+        return false;
     }
 
+
+    /**
+     * @dev Returns total no. of attendees.
+     */
+    function totalAttended() public view returns (uint) {
+        // if using maps then calculate based on them
+        if (0 < attendanceMaps.length) {
+            uint sum = 0;
+            for (uint i = 0; i < attendanceMaps.length; i++) {
+                uint map = attendanceMaps[i];
+                // brian kerninghan bit-counting method - O(log(n))
+                while (map != 0) {
+                    map &= (map - 1);
+                    sum++;
+                }
+            }
+            // since maps can contain more bits than there are registrants, we cap the value!
```

Comment:
Does this happen all the time or rather people can put that unintentionally? The logic is fine if former but better to throw if the latter.

Reply:
Actually it will only happen if someone lazily sets extra bits to 1 in the final map value. But again, I don't want to throw since it's a getter and we're not changing state - safer to just cap the value.

Reply:
Even though this function itself is getter, it's used in `payout` function which is state changing. Isn't it better to throw in any incorrect data rather than returning false reply? Also will this check and throw when `attendanceMaps` gets set (which is probably more important than throwing here)?

Reply:
Yeah, finalize() is where this check should be.

Reply:
On balance, I think you're right that finalize() should check before setting `attendanceMaps` but I don't think the extra gas usage is worth the effort. Much cheaper to simply do this check here.